cmake/pcl_find_vtk.cmake

```diff
@@ -102,27 +102,25 @@ if (vtkMissingComponents)
   message(WARNING "Missing vtk modules: ${vtkMissingComponents}")
 endif()
 
-if(WITH_QT)
-  if(VTK_VERSION VERSION_LESS 9.0)
-    if(";${VTK_MODULES_ENABLED};" MATCHES ";vtkGUISupportQt;" AND ";${VTK_MODULES_ENABLED};" MATCHES ";vtkRenderingQt;")
-      set(HAVE_QVTK TRUE)
-      #PCL_VTK_COMPONENTS is used in the PCLConfig.cmake to refind the required modules.
-      #Pre vtk 9.0, all vtk libraries are linked into pcl_visualizer.
-      #Subprojects can link against pcl_visualizer and directly use VTK-QT libraries.
-      list(APPEND PCL_VTK_COMPONENTS vtkRenderingQt vtkGUISupportQt)
-    else()
-      unset(HAVE_QVTK)
-    endif()
+if(VTK_VERSION VERSION_LESS 9.0)
+  if(";${VTK_MODULES_ENABLED};" MATCHES ";vtkGUISupportQt;" AND ";${VTK_MODULES_ENABLED};" MATCHES ";vtkRenderingQt;")
+    set(HAVE_QVTK TRUE)
+    #PCL_VTK_COMPONENTS is used in the PCLConfig.cmake to refind the required modules.
+    #Pre vtk 9.0, all vtk libraries are linked into pcl_visualizer.
+    #Subprojects can link against pcl_visualizer and directly use VTK-QT libraries.
+    list(APPEND PCL_VTK_COMPONENTS vtkRenderingQt vtkGUISupportQt)
   else()
-	if(";${VTK_AVAILABLE_COMPONENTS};" MATCHES ";GUISupportQt;" AND ";${VTK_AVAILABLE_COMPONENTS};" MATCHES ";RenderingQt;")
-      set(HAVE_QVTK TRUE)
-      #PCL_VTK_COMPONENTS is used in the PCLConfig.cmake to refind the required modules.
-      #Post vtk 9.0, only required libraries are linked against pcl_visualizer.
-      #Subprojects need to manually link to VTK-QT libraries.
-      list(APPEND PCL_VTK_COMPONENTS RenderingQt GUISupportQt)
-    else()
-      unset(HAVE_QVTK)
-    endif()
+    unset(HAVE_QVTK)
+  endif()
+else()
+  if(";${VTK_AVAILABLE_COMPONENTS};" MATCHES ";GUISupportQt;" AND ";${VTK_AVAILABLE_COMPONENTS};" MATCHES ";RenderingQt;")
+    set(HAVE_QVTK TRUE)
+    #PCL_VTK_COMPONENTS is used in the PCLConfig.cmake to refind the required modules.
+    #Post vtk 9.0, only required libraries are linked against pcl_visualizer.
+    #Subprojects need to manually link to VTK-QT libraries.
+    list(APPEND PCL_VTK_COMPONENTS RenderingQt GUISupportQt)
+  else()
+    unset(HAVE_QVTK)
   endif()
 endif()
```

Comment:
Maybe deduplicate `unset(HAVE_QVTK)`

```cmake
if(VTK_VERSION VERSION_LESS 9.0 AND ";${VTK_MODULES_ENABLED};" MATCHES ";vtkGUISupportQt;" AND ";${VTK_MODULES_ENABLED};" MATCHES ";vtkRenderingQt;")
  set(HAVE_QVTK TRUE)
  #PCL_VTK_COMPONENTS is used in the PCLConfig.cmake to refind the required modules.
  #Pre vtk 9.0, all vtk libraries are linked into pcl_visualizer.
  #Subprojects can link against pcl_visualizer and directly use VTK-QT libraries.
  list(APPEND PCL_VTK_COMPONENTS vtkRenderingQt vtkGUISupportQt)
elseif(";${VTK_AVAILABLE_COMPONENTS};" MATCHES ";GUISupportQt;" AND ";${VTK_AVAILABLE_COMPONENTS};" MATCHES ";RenderingQt;")
  set(HAVE_QVTK TRUE)
  #PCL_VTK_COMPONENTS is used in the PCLConfig.cmake to refind the required modules.
  #Post vtk 9.0, only required libraries are linked against pcl_visualizer.
  #Subprojects need to manually link to VTK-QT libraries.
  list(APPEND PCL_VTK_COMPONENTS RenderingQt GUISupportQt)
else()
  unset(HAVE_QVTK)
endif()
```

Reply:
Maybe remove it all together, since its unlikely that you would remove libraries between two consecutive Cmake runs?

Reply:
Could be risky. E.g. when you VTK 7 with Qt and switch to VTK 9, but forgot to compile the Qt module, `HAVE_QVTK` will be still true.

Reply:
Yes, if you don't rerun from scratch thats a possible error case.

Reply:
`VTK_VERSION VERSION_LESS 9.0 AND ` In your suggestion could be removed as well.
```
if(old name matches)
elseif(new name matches)
else
  unset it
endif
```
?

Reply:
indeed ;-)